### PR TITLE
[TCFMM-14] Oracles tests

### PR DESCRIPTION
**Pull request by Martoon-00**
[//]: # (This is a template of a pull request template.)
[//]: # (You should modify it considering specifics of a particular repository and put it there.)
[//]: # (Comments like this are meta-comments, they shouldn't be present in the final template.)
[//]: # (Comments starting with '

Related to TCFMM-14

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [Specification](../tree/master/docs/specification.md)
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

[//]: # (Update link to style guide if necesary or remove if it's not present)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/types.mligo [hunk 1]
```diff
@@ -268,12 +268,12 @@ type timed_cumulatives_buffer = {
 }
 
 let init_cumulatives_buffer (extra_reserved_slots : nat) : timed_cumulatives_buffer =
```
Comment (pasqu4le) [inline ligo/types.mligo:270]: Come to think of it, this may be better located into `defaults.mligo`.

File: haskell/test/Test/TimedOracles.hs [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    advanceLevel 3
+
+    checkedTimes <- do
+      cumulatives <- getStorage cfmm >>= lastRecordedCumulatives
+      let time = tcTime cumulatives
+      return $ timestampPlusSeconds time <$> [-3 .. 0]
+
+    -- Our property of interest here:
+    -- lim{t -> record_time} cumulative(t) = cumulative(record_time)
+    -- We will also check places of regular growth for the company.
```
Comment (pasqu4le) [inline haskell/test/Test/TimedOracles.hs:55]: WDYM by "for the company" here?

File: haskell/test/Test/TimedOracles.hs [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+      CumulativesBufferRPC{..} <- sCumulativesBufferRPC <$> getStorage cfmm
+      cbReservedLengthRPC @== 1 + incr
+      cbFirstRPC @== 1
+      cbLastRPC @== 1
+
+      runInvariantsChecks
+
+    -- We already registered one value, so performing one less probe
+    storageSnapshots1 <- for [1 .. incr - 1] \i -> do
+      -- Aribtrary call to put a new record in the buffer
```
Comment (pasqu4le) [inline haskell/test/Test/TimedOracles.hs:132]: nitpick: `Aribtrary` -> `Arbitrary`

note: there's at least one more place with this typo

File: haskell/test/Test/TimedOracles.hs [hunk 4]
```diff
@@ -0,0 +1,355 @@
+-- SPDX-FileCopyrightText: 2021 Arthur Breitman
+-- SPDX-License-Identifier: LicenseRef-MIT-Arthur-Breitman
+
+module Test.TimedOracles
```
[the remark this answers is not in the document]
Comment (dcastro) [reply haskell/test/Test/TimedOracles.hs:4]: Just merged #40

File: haskell/test/Test/TimedOracles.hs (hunk 2 above)
Comment (Martoon-00) [reply haskell/test/Test/TimedOracles.hs:62]: I meant "along with that", the current wording is really confusing :unamused:

**Pull request merged**